Fix insert_by_period for Postgres

LMandtler opened the pull request:
The insert_by_period materialization does not work for a postgres target.

Earlier discussions on this topic:

1. https://discourse.getdbt.com/t/a-materialization-for-problematic-initial-builds/60/7
2. https://github.com/dbt-labs/dbt-labs-experimental-features/issues/32

This is fixed in this PR by taking the ideas of the first link above.
Disclaimer: These are not my ideas. I am just a regular user of this materialization and am happy to have it fixed.

leonard-mandtler committed:
fix dependency on db_utils
The insert_by_period materialization depends on dbt_utils getdate() functions. So to get insert_by_period work, dbt_utils must be added to the packages.

leonard-mandtler committed:
exchange getdate() for dbt_utils.current_timestamp()
The getdate() function is not valid across different warehouses. Therefore, we need to use the corresponding dbt_utils function

leonard-mandtler committed:
alias for subquery
Postgres needs subqueries to have an alias

leonard-mandtler committed:
Remove schema name for postgres
In the creation of the relation node, Postgres cannot work with a schema. Therefore I created a small macro, where I use dispatch to differentiate between postgres and the other warehouses.

LMandtler commented:
The fixes for Postgres have now been implemented and tested against a postgres instance. However, I had no chance to test against any other backends. Reviews are very welcome.

joellabes commented on insert_by_period/macros/get_period_boundaries.sql:
```diff
@@ -12,7 +12,7 @@
                                 -1,
                                 "nullif('" ~ stop_date ~ "','')::timestamp") }},
-            getdate()
+            {{ dbt_utils.current_timestamp() }}
```
In dbt v1.3, this changed to `dbt.current_timestamp()` (note `dbt` not `dbt_utils`), and in dbt utils 1.0 (coming very very soon) `current_timestamp()` is being removed. I recommend not creating a dependency on utils and instead using the one provided in the `dbt` namespace directly. 

Note that this would create a requirement of dbt Core 1.3 for this project, which you can enforce by adding a require-dbt-version block https://docs.getdbt.com/reference/project-configs/require-dbt-version

leonard-mandtler replied:
Thanks for pointing me to this. This is resolved now.

joellabes reviewed (changes requested):
Looks good! I recommend not creating a dependency on dbt utils as the timestamp code has just been moved out and this will cause issues. Other than that excited to get this merged 🎉

leonard-mandtler committed:
remove dependency on dbt_utils

leonard-mandtler committed:
add requirement for dbt>=1.3.0
This is needed to resolve dbt.current_timestamp() which was introduced in 1.3.0

joellabes commented:
Thanks @LMandtler!
